secure_message/common/utilities.py

```diff
@@ -143,22 +143,39 @@ def get_details_by_uuids(uuids):
 
 def add_to_and_from_details(messages):
     """Adds user details for sender and recipient"""
+    for message in messages:
+        message['@msg_from'] = _get_from_details(message)
+        message['@msg_to'] = _get_to_details(message)
+    return messages
 
-    uuid_list = []
 
-    for message in messages:
-        uuid_list.extend([uuid for uuid in message['msg_to'] if uuid not in uuid_list])
-        if message['msg_from'] not in uuid_list:
-            uuid_list.append(message['msg_from'])
+def _get_from_details(message):
+    """looks up the details for the from users"""
+    if message['from_internal']:
+        from_details, _ = internal_user_service.get_user_details(message['msg_from'])
```

Comment:
Get user details shouldn't be returning a tuple - it's not used anywhere and makes the code overly verbose. If you need to handle HTTPErrors deal with them in the calling code or let the exception move up the stack

Reply:
In the first comment its logging twice , the lower and where you see it . Removing the higher one along with the tuple return

Reply:
Also changing party and user lookups to return None in the case of no data and log errors closer to the error detection.